src/vanilla.ts

```diff
@@ -12,7 +12,7 @@ export interface Subscribe<T extends State> {
   (listener: StateListener<T>): () => void
   <StateSlice>(
     listener: StateSliceListener<StateSlice>,
-    selector: StateSelector<T, StateSlice>,
+    selector?: StateSelector<T, StateSlice>,
```

Comment:
Hm, but if this works, we don't need L12?
What will be `StateSlice` inferred if you don't specify the selector?

If there's a better way of typing subscribe, it's very welcomed. Would you explain it in detail?

Reply:
The reason for this change is that I wrap the subscribe function and allow users to optionally provide their own selector. So I am passing in a parameter that can be undefined (as a result, line 12 does not apply). 

This typing change enables this type of workflow:

```ts
function subscribeToStore(store: StoreApi<any>, callback: (selectedState) => void, selector?: (state) => any) {
    // this causes an error without the proposed change
    store.subscribe(callback, selector);
}

Reply:
And now that you mention it, line 12 seems unnecessary.

Reply:
Ok now I actually understand what this line was doing:

```ts
<StateSlice>(listener: StateSliceListener<StateSlice>, selector: StateSelector<T, StateSlice>, equalityFn?: EqualityChecker<StateSlice>): () => void;
```

It's not very common TS to infer a generic like that, so had to wrap my head around it.

I revised my fix to preserve the type inference while still enabling more dynamic workflows like the one I proposed.

Rather than change the existing types the interface can map to, I've added new one:

```ts
(listener: StateListener<any>, selector?: StateSelector<T, any>, equalityFn?: EqualityChecker<any>): () => void;
```

Basically the "do whatever you like" type. Everything works exactly as it is now, but if you pass a potentially undefined type as the selector, TS compiles, but you give up type assistance.

I wasn't sure if I should use `unknown` or `any`, but decided to be more permissive here to avoid annoying users that "know what they're doing". Let me know if you prefer `unknown` over `any`.